thunder/core/module.py

```diff
@@ -187,7 +187,10 @@ def __getattr__(self, name: str) -> Any:
 
     def state_dict(self, *args: Any, **kwargs: Any) -> Any:
         # this is broken for transformed modules!!!
-        return self._model.state_dict(*args, **kwargs)
+        state_dict = self._model.state_dict(*args, **kwargs)
+        for transform in self._lc_transforms:
```

Comment:
I think the transforms should be iterated in reverse.
```suggestion
        for transform in reverse(self._lc_transforms):
```

Reply:
why?

Reply:
So the transform_module / transform_state_dict_for_submodule is applied in the order of the transforms. This is a sort of inverse transformation.
Say (not working yet) you have a quantization and a fsdp transform.
Then in load_original_state_dict you have
```
sd for untransformed ("original" module), unquantized unsharded
--> quant_trafo.transform_state_dict_for_submodule
sd for quantized, unsharded module
--> fsdp_trafo.transform_state_dict_for_submodule
sd for quantized, sharded module
```
Now the `original_state_dict` would need to  go the other way:
```
sd for quantized, sharded module
--> fsdp_trafo.reverse_transform_state_dict_for_submodule
sd for quantized, unsharded module
--> quant_trafo.reverse_transform_state_dict_for_submodule
sd for untransformed ("original" module), unquantized unsharded
```
for this to work, we would need the reverse order of the transforms.